pkgs/development/compilers/gcc/9/default.nix

```diff
@@ -158,34 +158,31 @@ stdenv.mkDerivation ({
   inherit noSysDirs staticCompiler crossStageStatic
     libcCross crossMingw;
 
-  depsBuildBuild = [ buildPackages.stdenv.cc ];
-  nativeBuildInputs = [ texinfo which gettext ]
-    ++ (optional (perl != null) perl)
-    ++ (optional langAda gnatboot)
-    # The builder relies on GNU sed (for instance, Darwin's `sed' fails with
-    # "-i may not be used with stdin"), and `stdenvNative' doesn't provide it.
-    ++ (optional buildPlatform.isDarwin gnused)
-    ;
-
-  # For building runtime libs
-  depsBuildTarget =
-    (
-      if hostPlatform == buildPlatform then [
-        targetPackages.stdenv.cc.bintools # newly-built gcc will be used
-      ] else assert targetPlatform == hostPlatform; [ # build != host == target
-        stdenv.cc
-      ]
-    )
-    ++ optional targetPlatform.isLinux patchelf;
-
-  buildInputs = [
-    gmp mpfr libmpc
-    targetPackages.stdenv.cc.bintools # For linking code at run-time
-  ] ++ (optional (isl != null) isl)
-    ++ (optional (zlib != null) zlib)
+  inherit (import ../common/dependencies.nix {
+    inherit
+      lib
```

Comment:
Naive question: can we also arredicate this inherit list?

Reply:
i tried passing `@args` but https://nixos.org/manual/nix/stable/language/constructs.html?highlight=%40args#functions

![image](https://user-images.githubusercontent.com/56650223/217331760-78a8ab24-3e65-493b-b2df-f42d7051ebd0.png)

Reply:
I think the right way to do this is `lib.newScope`.